mon-pix/app/services/locale.js

```diff
@@ -2,11 +2,20 @@ import Service, { inject as service } from '@ember/service';
 import config from 'mon-pix/config/environment';
 
 const { COOKIE_LOCALE_LIFESPAN_IN_SECONDS } = config.APP;
+export const DEFAULT_LOCALE = 'fr';
+export const FRENCH_INTERNATIONAL_LOCALE = 'fr';
+export const ENGLISH_INTERNATIONAL_LOCALE = 'en';
 
 export default class LocaleService extends Service {
   @service cookies;
   @service currentDomain;
 
+  handleSupportedLanguage(language) {
+    if (!language) return;
+    const supportedLanguages = [FRENCH_INTERNATIONAL_LOCALE, ENGLISH_INTERNATIONAL_LOCALE];
```

Comment:
2 questions par rapport à cette ligne : 
1. Pour définir les locales supportées dans ce service alors qu'on a les locales supportées en appelant le service `intl` d'Ember : `this.intl.get('locales')` ? Voilà comment c'est utilisé dans Pix Orga : https://github.com/1024pix/pix/blob/dev/orga/app/services/session.js#L34
2. Si on veut vraiment définir cette constante, alors il faudrait mettre cette constante `const supportedLanguages = [FRENCH_INTERNATIONAL_LOCALE, ENGLISH_INTERNATIONAL_LOCALE]` dans le scope global du fichier, de manière à ne pas redéfinir cette constante chaque fois que la méthode est appelé.

Reply:
Je trouve que c'est plus parlant d'avoir un tableau pour les devs qui arrivent car on voit de suite qu'il n'y a que 2 locales sur le projet même si  je suis d'accord avec toi`this.intl.get('locales')` nous renverra la vérité de ce qu'il y vraiment comme locales.

J'ai pas compris ta remarque `dans le scope global du fichier` tu voulais dire le  scope global du projet ?

Reply:
Bonne remarque ! 
En effet, si les locales sont disponibles avec le service `intl`, il n'y a pas besoin de les redéfinir.  

Je fais la modification 😄

Reply:
Pour la remarque sur le scope voici ci-dessous ce que donnerait le code du fichier si on remonte la constante au niveau du scope global du fichier.

Si on ne fait pas ça la variable `supportedLanguages` est redéfinie inutilement à chaque appel de la fonction.

Enfin, remonter des constantes dont on connait déjà la valeur avant l'instantiation du code en mémoire met en évidence les valeurs « magiques » et les constantes qui pourraient être factorisées entre plusieurs fichiers/modules.

```javascript
import Service, { inject as service } from '@ember/service';
import config from 'mon-pix/config/environment';

const { COOKIE_LOCALE_LIFESPAN_IN_SECONDS } = config.APP;
export const DEFAULT_LOCALE = 'fr';
export const FRENCH_INTERNATIONAL_LOCALE = 'fr';
export const ENGLISH_INTERNATIONAL_LOCALE = 'en';

const SUPPORTED_LANGUAGES = [FRENCH_INTERNATIONAL_LOCALE, ENGLISH_INTERNATIONAL_LOCALE];

export default class LocaleService extends Service {
  @service cookies;
  @service currentDomain;

  handleSupportedLanguage(language) {
    if (!language) return;
    return SUPPORTED_LANGUAGES.includes(language) ? language : DEFAULT_LOCALE;
  }

  // ...
}
```

J'ai fait un commentaire identique dans une PR récente ici : https://github.com/1024pix/pix-site/pull/525#discussion_r1166972046

Reply:
Pour le fait d'indiquer dans le code quelles sont les locales supportées, oui c'est plus explicite d'avoir des constantes dans notre service maison `this.locale`, mais dans ce cas cette information plus explicite n'apporte en fait pas de compréhension supplémentaire, car comme tu le dis, il est évident pour toutes les personnes travaillant sur le projet quelles sont les locales supportées (en regardant très simplement l'arborescence des fichiers).

Donc en lui-même cet ajout de la constante supportedLanguages/SUPPORTED_LANGUAGES n'est pas gênant, sans apporter de valeur ajoutée, par contre il apporte le petit inconvénient de désynchroniser un tout petit peu les codes de Pix Orga et Pix Certif qui se basent eux sur `this.intl.get('locales')`.

Donc c'est vraiment de l'ordre du détail.